src/notificationmanager.cpp

```diff
@@ -95,39 +133,51 @@ void NotificationManager::handleUpdateNotificationGroup(const QVariantMap notifi
     // If we have deleted notifications, we need to update possibly existing ones
     if (!removedNotificationIds.isEmpty() && !activeNotifications.isEmpty()) {
         LOG("Some removals happend, but we have" << activeNotifications.size() << "existing notifications.");
-        QVariantMap firstActiveNotification = activeNotifications.first().toMap();
-        activeNotifications.remove(firstActiveNotification.value("id").toString());
-        QVariantMap newFirstActiveNotification = this->sendNotification(chatId, firstActiveNotification, activeNotifications);
+        const QVariantMap firstActiveNotification = activeNotifications.first().toMap();
+        activeNotifications.remove(firstActiveNotification.value(ID).toString());
+        QVariantMap newFirstActiveNotification = this->sendNotification(chatId, firstActiveNotification, activeNotifications, needFeedback);
         QVariantMap newActiveNotifications;
         QListIterator<QVariant> activeNotificationsIterator(activeNotifications.values());
         while (activeNotificationsIterator.hasNext()) {
             QVariantMap newActiveNotification = activeNotificationsIterator.next().toMap();
-            newActiveNotification.insert("replaces_id", newFirstActiveNotification.value("replaces_id"));
-            newActiveNotifications.insert(newActiveNotification.value("id").toString(), newActiveNotification);
+            newActiveNotification.insert(REPLACES_ID, newFirstActiveNotification.value(REPLACES_ID));
+            newActiveNotifications.insert(newActiveNotification.value(ID).toString(), newActiveNotification);
         }
-        newActiveNotifications.insert(newFirstActiveNotification.value("id").toString(), newFirstActiveNotification);
+        newActiveNotifications.insert(newFirstActiveNotification.value(ID).toString(), newFirstActiveNotification);
         activeNotifications = newActiveNotifications;
     }
 
-    if (activeNotifications.isEmpty()) {
-        this->controlLedNotification(false);
-    }
-
-    QVariantList addedNotifications = notificationGroupUpdate.value("added_notifications").toList();
     QListIterator<QVariant> addedNotificationIterator(addedNotifications);
     while (addedNotificationIterator.hasNext()) {
-        QVariantMap addedNotification = addedNotificationIterator.next().toMap();
-        QVariantMap activeNotification = this->sendNotification(chatId, addedNotification, activeNotifications);
-        activeNotifications.insert(activeNotification.value("id").toString(), activeNotification);
+        const QVariantMap addedNotification = addedNotificationIterator.next().toMap();
+        const QVariantMap activeNotification = this->sendNotification(chatId, addedNotification, activeNotifications, needFeedback);
+        activeNotifications.insert(activeNotification.value(ID).toString(), activeNotification);
+    }
+
+    if (needFeedback) {
+        ngfClient->play(NGF_EVENT);
     }
```

Comment:
Why did you separate the NGF play event from the actual action to send a new notification? Seems to be inconsistent and possibly triggers too many bzzzzs...

Reply:
My intention was actually to reduce number of bzzz - it used to be inside `sendNotification` (which can be called more than once), moving bzzz outside can't trigger more, only less.

Reply:
Actually, I ended up moving it back to the method which actually posts the notifications...